src/Functions/DateTimeTransforms.h

```diff
@@ -46,6 +46,11 @@ namespace ErrorCodes
         throw Exception("Illegal type DateTime of argument for function " + std::string(name), ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
     }
 
+    static inline Int64 date32IsNotSupported(const char * name)
```

Comment:
This and above methods could be `[[noreturn]]`. Bonus points for moving their implementation into a to-be-created source file.

Reply:
I put no return this way: 
`[[noreturn]] static inline Int64 date32IsNotSupported(const char * name)`
for all 3 methods.